src/sd.rs

```diff
@@ -0,0 +1,301 @@
+use core::borrow::Borrow;
+
+use crate::spi::SpiDriver;
+use crate::sys::*;
+
+#[cfg(esp_idf_soc_sdmmc_host_supported)]
+use mmc::SdMmcHostDriver;
+use spi::SdSpiHostDriver;
+
+#[cfg(esp_idf_soc_sdmmc_host_supported)]
+pub mod mmc;
+pub mod spi;
+
+const _SDMMC_HOST_FLAG_SPI: u32 = 1 << 3;
+const _SDMMC_HOST_FLAG_DDR: u32 = 1 << 4;
+const _SDMMC_HOST_FLAG_DEINIT_ARG: u32 = 1 << 5;
+
+pub type SdCardConfiguration = config::Configuration;
+
+pub mod config {
+    #[cfg(not(any(
+        esp_idf_version_major = "4",
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+    )))] // For ESP-IDF v5.2 and later
+    use crate::sys::*;
```

Comment:
why this separate glob import only behind restricted version, if you do still have one in the top of the file.

Reply:
Because the one at the top of the file is for a different module.